photosynthesis_metrics/perceptual.py

```diff
@@ -0,0 +1,280 @@
+            2018 IEEE/CVF Conference on Computer Vision and Pattern Recognition
+            https://arxiv.org/abs/1801.03924
+
+        """
+        super().__init__()
+
+        if feature_extractor == "vgg16":
+            self.model = vgg16(pretrained=True, progress=False)
+            self.layers = [VGG16_LAYERS[l] for l in layers]
+        elif feature_extractor == "vgg19":
+            self.model = vgg19(pretrained=True, progress=False)
+            self.layers = [VGG19_LAYERS[l] for l in layers]
+        else:
+            raise ValueError("Unknown feature extractor")
+
+        if replace_pooling:
+            for i, layer in enumerate(self.model.features):
+                if isinstance(layer, torch.nn.MaxPool2d):
+                    self.model.features[i] = torch.nn.AvgPool2d(kernel_size=2, stride=2, padding=0)
+
+        # Disable gradients
+        for param in self.model.parameters():
+            param.requires_grad_(False)
+
+        self.distance = {
+            "mse": nn.MSELoss(reduction='none'),
+            "mae": nn.L1Loss(reduction='none'),
+        }[distance]
+
+        self.weights = [torch.tensor(w) for w in weights]
+        mean = torch.tensor([0.485, 0.456, 0.406]) if normalize_input else torch.tensor([0., 0., 0.])
+        std = torch.tensor([0.229, 0.224, 0.225]) if normalize_input else torch.tensor([1., 1., 1.])
+        self.mean = mean.view(1, 3, 1, 1)
+        self.std = std.view(1, 3, 1, 1)
+        
+        self.normalize_features = normalize_features
+        self.reduction = reduction
+
+    def forward(self, prediction: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+        r"""Computation of Content loss between feature representations of prediction and target tensors.
+
+        Args:
+            prediction: Tensor of prediction of the network.
+            target: Reference tensor.
+
+        """
+        _validate_input(input_tensors=(prediction, target), allow_5d=False)
+        prediction, target = _adjust_dimensions(input_tensors=(prediction, target))
+        return self.compute_metric(prediction, target)
+    
+    def compute_metric(self, prediction: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+        # Normalize input
+        mean, std = self.mean.to(prediction), self.std.to(prediction)
+        prediction = (prediction - mean) / std
+        target = (target - mean) / std
+
+        self.model.to(prediction)
+        prediction_features = self.get_features(prediction)
+        target_features = self.get_features(target)
+        
+        distances = [self.distance(x, y) for x, y in zip(prediction_features, target_features)]
+
+        # Scale distances, then average in spatial dimensions, then stack and sum in channels dimension
+        loss = torch.cat([(d * w.to(d)).mean(dim=[2, 3]) for d, w in zip(distances, self.weights)], dim=1).sum(dim=1)
+
+        # Solve big memory consumption
+        torch.cuda.empty_cache()
```

Comment:
Be careful with that. It may result in a huge performance degradation because GPU I/O frequently takes way more time then computations. 
Also note that this code actually effects the user of the library who may use this metric during training/inference. In this case it may lead not only in slow computation of the metric, but also performance degradation in the whole training/validation pipeline.

I would remove it.